pkg/cloudfoundry/ManifestUtils.go

```diff
@@ -0,0 +1,226 @@
+package cloudfoundry
+
+import (
+	"fmt"
+	"github.com/ghodss/yaml"
+	"github.com/pkg/errors"
+	"io/ioutil"
+	"reflect"
+
+	"github.com/SAP/jenkins-library/pkg/log"
+)
+
+const propApplications = "applications"
+const propBuildpacks = "buildpacks"
+const propBuildpack = "buildpack"
+
+// Manifest ...
+type Manifest struct {
+	self     map[string]interface{}
+	modified bool
+	name     string
+}
+
+var m Manifest
+
+var _readFile = ioutil.ReadFile
+
+// ReadManifest ...
+func ReadManifest(name string) (Manifest, error) {
+
+	log.Entry().Infof("Reading manifest file  '%s'", name)
+
+	m := Manifest{self: make(map[string]interface{}), name: name, modified: false}
+
+	content, err := _readFile(name)
+	if err != nil {
+		return m, errors.Wrapf(err, "cannot read file '%v'", m.name)
+	}
+
+	err = yaml.Unmarshal(content, &m.self)
+
+	if err != nil {
+		return m, errors.Wrapf(err, "Cannot parse yaml file '%s': %s", m.name, string(content))
+	}
+
+	log.Entry().Infof("Manifest file '%s' has been parsed", m.name)
+
+	return m, nil
+}
+
+// WriteManifest ...
+func (m *Manifest) WriteManifest() error {
+
+	d, err := yaml.Marshal(&m.self)
+	if err != nil {
+		return err
+	}
+
+	log.Entry().Debugf("Writing manifest file '%s'", m.name)
+	err = ioutil.WriteFile(m.name, d, 0644)
+
+	if err == nil {
+		m.modified = false
+	}
+
+	log.Entry().Debugf("Manifest file '%s' has been written", m.name)
+	return err
+}
+
+// GetName ...
+func (m Manifest) GetName() string {
+	return m.name
+}
+
+// GetApplications ...
```

Comment:
It returns the applications contained in the manifest. Having method like that documented or not is an old question. The more we talk about official and solid APIs the more it should be documented. On the other hand documenting the obvious does not make sense. The very best documentation is the source code. Maybe I saw too many cases where something we properly and with a lot of passion documented initially, but afterwards the coding moved slightly without adjusting the documentation ... In case you still think it should be documented, please add a note again. In that case i will document ...

Reply:
IMO we are still close to "document the obvious", but OK I added documentation now. One good thing in that context is: up to now we returned an interface slice from getApplications. But it is clear that we have a map with a string as key and an interface as value. I improved that as an outcome of providing the docu.

Reply:
Thanks!